src/gui/DatabaseWidget.cpp

```diff
@@ -326,6 +327,20 @@ void DatabaseWidget::cloneEntry()
     return;
 }
 
+void DatabaseWidget::getTotp()
+{
+    Entry* currentEntry = m_entryView->currentEntry();
+    if (!currentEntry) {
+        Q_ASSERT(false);
+        return;
+    }
+
+    TotpDialog* totpDialog = new TotpDialog(this, currentEntry);
+    totpDialog->setWindowFlags(totpDialog->windowFlags() | Qt::WindowStaysOnTopHint);
+    totpDialog->show();
```

Comment:
I would prefer opening this as a modal dialog (sheet on OS X) with open(). Then you also don't need `Qt::WindowStaysOnTopHint`.